docroot/modules/custom/yukon_sso/yukon_sso.module

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * @file
+ * Main module file.
+ */
+
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_openid_connect_userinfo_save().
+ */
+function yukon_sso_openid_connect_userinfo_save($account, $context) {
+  $roles_mapping = yukon_sso_get_roles_mapping();
+
+  // SSO roles should be updated on each authorization.
+  // Delete all SSO user roles.
+  foreach ($account->getRoles() as $role) {
+    if (in_array($role, $roles_mapping)) {
+      $account->removeRole($role);
+    }
+  }
+
+  // Assign roles according to SSO response.
+  if (!empty($context['user_data']['urn:myyukon.ca/groups'])) {
+    foreach ($context['user_data']['urn:myyukon.ca/groups'] as $auth0_role) {
+      if (isset($roles_mapping[$auth0_role])) {
+        $account->addRole($roles_mapping[$auth0_role]);
+      }
+    }
+  }
+}
+
+/**
+ * SSO and Drupal roles mapping.
+ *
+ * @return array
+ *   Roles mapping list.
+ */
+function yukon_sso_get_roles_mapping(): array {
+  return [
+    'web-admin' => 'site_administrator',
+    'web-trans' => 'translator',
+    'web-team' => 'team_administrator',
+    'web-alerts' => 'author_in_page_alerts',
+    'web-pub' => 'publisher',
+    'web-editor' => 'editor',
+    'web-writer' => 'writer',
```

Comment:
@leeomara Is SSO roles token correct? Because on testing I got "web-writer" in the response.
But on the auth0 UI I see "web-yukon-writer". What is the correct one?

Reply:
The correct names are all `web-yukon-*`.